src/pipelines/covariates/pipeline.py

```diff
@@ -0,0 +1,13 @@
+from typing import Any, Dict
+from pandas import DataFrame
+
+from lib.time import date_offset
+from lib.default_pipeline import DefaultPipeline
+
+class CovariatesPipeline(DefaultPipeline):
```

Comment:
Please sync with the latest commit, there's been a few breaking changes but I think future changes will be much smaller. Let me know if you run into issues converting this into a `PipelineChain`, it should be very straight-forward.